crates/polars-plan/src/dsl/string.rs

```diff
@@ -526,6 +526,26 @@ impl StringNameSpace {
         )
     }
 
+    /// Take the first `n` characters of the string values.
+    pub fn head(self, n: Expr) -> Expr {
```

Comment:
I'm wondering if we can't just define these in terms of a `slice` operation - that would save a lot of code bloat. But that might not work with negative indices.

Reply:
Hi @stinodego -- this was my initial intent, when I said I would piggyback on @reswqa's implementation of `str.slice`. I realized soon after that the negative indexing for head requires calculation of the string length to determine the end of the slice.

Here are the operations and their `slice` equivalents:

```python
s.str.head(3)   # s.str.slice(3, None)
s.str.head(-3)  # no equivalent: must know string length for start offset

s.str.tail(3)   # s.str.slice(-3, None)
s.str.tail(-3)  # s.str.slice(3, None)
```

So for `tail`, we *could* do it because `slice` can run to the end of the string by itself, but for head we have no recourse. I suppose this would save us a little bit of bloat but it does make the code a bit asymmetric, but on the other hand the `tail` implementation is a bit more performant than `slice` because we have one fewer parameters, and so we have more fast paths. So it's a tradeoff here. What do you think?

Reply:
I guess you can use `slice` in combination with `len_chars`, but clearly it will be a bit more efficient to have a dedicated implementation like in this PR. I'll leave it to Ritchie to be the judge here.